assembler/include/shrimp/assembler.hpp

```diff
@@ -216,6 +237,24 @@ class Assembler final {
         }
     }
 
+    StrId parseStrId(std::string str)
```

Comment:
It doesn't looks like parsing subroutine, as it performs lookup in assembler table. Thus, name is not suitable.
Also, please add a check for end iterator.

Reply:
Also, don't copy string on call:
```suggestion
    StrId parseStrId(const char *str)
```

And, maybe, use transparent hashing: https://en.cppreference.com/w/cpp/container/unordered_map/find